src/main/java/jenkins/branch/MultiBranchProject.java

```diff
@@ -653,9 +653,35 @@ private void scheduleBuild(BranchProjectFactory<P, R> factory, final P item, SCM
                     name, getDisplayName());
             return;
         }
```

Comment:
also please rename argument `Cause[] cause` to `causes` (events already passing multiple causes here)